core/src/main/java/org/apache/iceberg/SchemaParser.java

```diff
@@ -197,7 +197,12 @@ private static Type typeFromJson(JsonNode json) {
 
   private static Literal<?> defaultFromJson(String defaultField, Type type, JsonNode json) {
     if (json.has(defaultField)) {
-      return Expressions.lit(SingleValueParser.fromJson(type, json.get(defaultField)));
```

Comment:
This is a problem for 1.10? Should we block the release on this? While I know the chances are unlikely here but if anyone is trying to set defaults close to Epoch they won't Overflow and will instead write incorrect values